Automattic/wp-calypso
# Jetpack Cloud: Social connections screen

#### Changes proposed in this Pull Request

For the new Jetpack Social plugin we need a version of the `/marketing/connections` screen that just handles the management of Publicize connections. This PR adds the framework for that, reusing the components from that screen.

Some things still to do/investigate/decide are:

- ~In creating this I tried to make the main component a new `section` under `client/jetpack-cloud` but when the `marketing/connections` components were imported there were errors related to lazy loading of components. Moving it to under `my-sites` fixed the issue, but I'm not sure that's the best solution.~ This is fixed now. I think it was related to how I was importing the CSS
- ~I was having a problem with deactivation of connections, but I think it was related to my test site being unavailable. This probably needs more testing to be sure.~ I've not been able to replicate this
- ~There are some permissions to work out. I think this section will be available in Calypso, so we should redirect to `/marketing/connections` if it's not Jetpack Cloud. Also people should have `publish_posts` permissions in order to manage connections.~ I've replicated the permissions from `marketing/connections`, and implemented a redirect.
- The CSS is somewhat hacky. We're importing some from the marketing section and then adding a couple of overrides. There is more to do here with button styles, but I think it's acceptable for now.

#### Testing instructions

- Use the live branch, or bring up Jetpack cloud locally with `yarn install && yarn start-jetpack-cloud`
- Head to `/jetpack-social` and select a Jetpack site
- Check that you can view, connect and disconnect your Publicize connections, as you would expect.

<img width="1028" alt="image" src="https://user-images.githubusercontent.com/96462/163474967-4455e9fd-bc9f-4fa3-9c4f-8268f15da3ae.png">
